hunts/src/puzzlesSlice.js

```diff
@@ -0,0 +1,102 @@
+import {
+  createSelector,
+  createSlice,
+  createAsyncThunk,
+  createEntityAdapter,
+} from "@reduxjs/toolkit";
+import api from "./api";
+
+export const deletePuzzle = createAsyncThunk(
+  "puzzles/deletePuzzle",
+  async ({ huntId, id }) => {
+    const response = await api.deletePuzzle(huntId, id);
+    return id;
+  }
+);
+
+export const fetchPuzzles = createAsyncThunk(
+  "puzzles/fetchPuzzles",
+  async (huntId) => {
+    const response = await api.getPuzzles(huntId);
+    return response;
+  }
+);
+
+function puzzleComparator(a, b) {
+  // Solved puzzles should appear below unsolved ones
+  if (a.status == "SOLVED" && b.status != "SOLVED") {
+    return 1;
+  } else if (b.status == "SOLVED" && a.status != "SOLVED") {
+    return -1;
+  }
+  // Feeders before metas
```

Comment:
Should feeders really come before metas? It seems more reasonable for feeders to be listed after the metas:

meta
   feeder1
   feeder2

etc.

Reply:
This is for sorting puzzles at the same depth of the tree, so if `feeder1` and `feeder2` are assigned to `meta`, they'll appear below `meta` as one of the children. I think if `feeder1` and `feeder2` are the same depth as `meta` then the mostly likely thing is that they're new puzzles that haven't been assigned a meta or whose meta hasn't been unlocked yet, and in that case it makes sense to me to keep them above the meta block so that they're more visible (for example imagine if the subtree of `meta` was 100 puzzles).